src/txn/txn_recover.c

```diff
@@ -482,11 +482,17 @@ __recovery_set_checkpoint_snapshot(WT_SESSION_IMPL *session)
     conn = S2C(session);
     counter = 0;
 
+    /* Initialize the recovery checkpoint snapshot variables to default values. */
+    conn->recovery_ckpt_snap_min = WT_TXN_NONE;
+    conn->recovery_ckpt_snap_max = WT_TXN_NONE;
+    conn->recovery_ckpt_snapshot_count = 0;
+
     /*
      * WiredTiger versions 10.0.1 onward have a valid checkpoint snapshot on-disk. Ignore reading
-     * the on-disk checkpoint snapshot from older versions.
+     * the on-disk checkpoint snapshot from older versions. WiredTiger backup doesn't contain the
+     * WiredTiger.turtle file. Allow retrieving the saved checkpoint snapshot if it exists.
      */
```

Comment:
Do you think it would be safer if we add `conn->recovery_major == 0` to this check as well?
https://github.com/wiredtiger/wiredtiger/blob/67e77c4fe970124e2c343c68596e9d39db265fae/src/txn/txn_recover.c#L990 
Something like this maybe?
`if (conn->unclean_shutdown && (conn->recovery_major == 0 || conn->recovery_major == 10) && conn->recovery_minor == 0 && conn->recovery_patch == 0)`

Reply:
I don't think that's necessary. Other WiredTiger major versions never generated a checkpoint snapshot. I've pushed a comment that I think covers this case.